drivers/espi/espi_it8xxx2.c

```diff
@@ -0,0 +1,1284 @@
+#define IT8XXX2_ESPI_MAX_FREQ_MASK GENMASK(2, 0)
+#define IT8XXX2_ESPI_CAPCFG1_MAX_FREQ_20 0
+#define IT8XXX2_ESPI_CAPCFG1_MAX_FREQ_25 1
+#define IT8XXX2_ESPI_CAPCFG1_MAX_FREQ_33 2
+#define IT8XXX2_ESPI_CAPCFG1_MAX_FREQ_50 3
+#define IT8XXX2_ESPI_CAPCFG1_MAX_FREQ_66 4
+
+#define IT8XXX2_ESPI_PC_READY_MASK  BIT(1)
+#define IT8XXX2_ESPI_VW_READY_MASK  BIT(1)
+#define IT8XXX2_ESPI_OOB_READY_MASK BIT(1)
+#define IT8XXX2_ESPI_FC_READY_MASK  BIT(1)
+
+#define IT8XXX2_ESPI_INTERRUPT_ENABLE          BIT(7)
+#define IT8XXX2_ESPI_TO_WUC_ENABLE             BIT(4)
+#define IT8XXX2_ESPI_VW_INTERRUPT_ENABLE       BIT(7)
+#define IT8XXX2_ESPI_INTERRUPT_PUT_PC          BIT(7)
+
+#define IT8XXX2_ESPI_UPSTREAM_INTERRUPT_ENABLE BIT(5)
+#define IT8XXX2_ESPI_UPSTREAM_CHANNEL_DISABLE  BIT(2)
+#define IT8XXX2_ESPI_UPSTREAM_DONE             BIT(1)
+
+#define IT8XXX2_ESPI_PUT_OOB_STATUS            BIT(7)
+#define IT8XXX2_ESPI_PUT_OOB_INTERRUPT_ENABLE  BIT(7)
+
+#define IT8XXX2_ESPI_INPUT_PAD_GATING          BIT(6)
+
+struct espi_it8xxx2_config {
+	uintptr_t base_espi_slave;
+	uintptr_t base_espi_vw;
+	uintptr_t base_espi_queue1;
+	uintptr_t base_espi_queue0;
+	uintptr_t base_ec2i;
+	uintptr_t base_kbc;
+	uintptr_t base_pmc;
+};
+
+struct espi_it8xxx2_data {
+	sys_slist_t callbacks;
+};
+
+/* Driver convenience defines */
+#define DRV_CONFIG(dev) ((const struct espi_it8xxx2_config *)(dev)->config)
+#define DRV_DATA(dev) ((struct espi_it8xxx2_data *)(dev)->data)
+
+struct vw_channel_t {
+	uint8_t  vw_index;      /* VW index of signal */
+	uint8_t  level_mask;    /* level bit of signal */
+	uint8_t  valid_mask;    /* valid bit of signal */
+};
+
+struct vwidx_isr_t {
+	void (*vwidx_isr)(const struct device *dev, uint8_t update_flag);
+	uint8_t vw_index;
+};
+
+enum espi_ch_enable_isr_type {
+	DEASSERTED_FLAG = 0,
+	ASSERTED_FLAG = 1,
+};
+
+struct espi_isr_t {
+	void (*espi_isr)(const struct device *dev, bool enable);
+	bool asserted_flag;
+};
+
+struct espi_vw_signal_t {
+	enum espi_vwire_signal signal;
+	void (*vw_signal_isr)(const struct device *dev);
+};
+
+/* EC2I bridge and PNPCFG devices */
+static const struct ec2i_t kbc_settings[] = {
+	/* Select logical device 06h(keyboard) */
+	{HOST_INDEX_LDN, LDN_KBC_KEYBOARD},
+	/* Set IRQ=01h for logical device */
+	{HOST_INDEX_IRQNUMX, 0x01},
+	/* Configure IRQTP for KBC. */
+	/*
+	 * Interrupt request type select (IRQTP) for KBC.
+	 * bit 1, 0: IRQ request is buffered and applied to SERIRQ
+	 *        1: IRQ request is inverted before being applied to SERIRQ
+	 * bit 0, 0: Edge triggered mode
+	 *        1: Level triggered mode
+	 *
+	 * This interrupt configuration should the same on both host and EC side
+	 */
+	{HOST_INDEX_IRQTP, 0x02},
+	/* Enable logical device */
+	{HOST_INDEX_LDA, 0x01},
+
+	/* Select logical device 05h(mouse) */
+	{HOST_INDEX_LDN, LDN_KBC_MOUSE},
+	/* Set IRQ=0Ch for logical device */
+	{HOST_INDEX_IRQNUMX, 0x0C},
+	/* Enable logical device */
+	{HOST_INDEX_LDA, 0x01},
```

Comment:
I didn't guard this device with a option is because enum lpc_peripheral_opcode listed E8042_WRITE_MB_CHAR without condition. I also checked other socs' driver, they didn't guard mouse device neither.
The device won't appear from system after booting if we skip handling mouse data from host (Don't enable CONFIG_8042_AUX option).

Reply:
> Can you guard this with a Kconfig option? For the legacy cros-ec, mouse support is only enabled for the ITE eval boards.

The Kconfig option created.